R/emissions_profile_any_compute_profile_ranking.R

```diff
@@ -73,26 +58,45 @@ flat_benchmarks <- function(data) {
 }
 
 add_rank <- function(data, .by) {
-  if (identical(.by, "all") | (identical(.by, "unit"))) {
-    if (identical(.by, "all")) .by <- NULL
-    mutate(
-      data,
-      profile_ranking = rank_proportion(.data[[extract_name(data, aka("co2footprint"))]]),
-      .by = all_of(.by)
-    )
-  } else {
-    if ((identical(.by, "isic_4digit")) | (identical(.by, c("unit", "isic_4digit")))) {
-      filtered_df <- data %>% filter(str_length(.data[["isic_4digit"]]) %in% c(NA_character_, 4, 5))
-      df <- anti_join(data, filtered_df, by = c("isic_4digit"))
-    } else {
-      filtered_df <- data %>% filter(is.na(.data[["tilt_sector"]]))
-      df <- anti_join(data, filtered_df, by = c("tilt_sector"))
-    }
-    output <- mutate(
-      df,
-      profile_ranking = rank_proportion(.data[[extract_name(data, aka("co2footprint"))]]),
-      .by = all_of(.by)
-    )
-    bind_rows(output, filtered_df)
-  }
```

Comment:
@maurolepore Please note that I am ranking only the co2 values which are not NA after excluding them using the `filtered_df` dataframe, and then concatenating it in the final output using `bind_rows`. 

And what you are doing with the new code is that you are first ranking all of them and then assigning the NA values to the ones we need. The NA cases should also not be used while ranking the right cases. For example: 

Your current code change: 
isic_4digit: ["12", "4444", "4444", "4444"].   (have to exclude "12" isic)
co2 values: [4, 5, 5, 6]
Rank: [NA, 0.66, 0.66, 1.0]

My previous code: 
isic_4digit: ["12", "4444", "4444", "4444"].   (have to exclude "12" isic)
co2 values: [4, 5, 5, 6]
Rank: [NA, 0.5, 0.5, 1.0]

From the above case you can understand that the 2-digit ISIC should be first excluded for ranking and then needs to be assigned an NA. Please let me know if its clear with this example or not! Thanks!

Reply:
Thanks @kalashsinghal, OK, I'll make the change and turn your example here into a test. BTW, this is an R package so I'd appreciate examples in R :-)